jp2_pc/Source/Trespass/ctrls.cpp

```diff
@@ -1069,28 +1069,42 @@ BOOL CUIListbox::LButtonUp(int x, int y, UINT keyFlags)
     return TRUE;
 }
 
+void CUIListbox::ScrollUp()
+{
+    if (int iNewTop = m_iTop - 1; iNewTop >= 0)
+    {
+        m_iTop = iNewTop;
+        m_bUpdate = TRUE;
+    }
+}
```

Comment:
Bearing in mind I'm not really a programmer and don't know what the general preference for this type of thing is, using an extra variable here seems a bit obtuse; is there a reason not to implement this as follows?
```cpp
void CUIListbox::ScrollUp()
{
    if ( (m_iTop - 1) >= 0 )
    {
        m_iTop--;
        m_bUpdate = TRUE;
    }
}
```

Reply:
That is indeed possible. The suggestion has been implemented.